dotnet/samples/GettingStartedWithAgents/AzureAIAgent/Step08_AzureAIAgent_Declarative.cs

```diff
@@ -0,0 +1,135 @@
+            model:
+              id: gpt-4o-mini
+            tools:
+              - type: openapi
+                name: RestCountriesAPI
+                description: Web API version 3.1 for managing country items, based on previous implementations from restcountries.eu and restcountries.com.
+                schema: '{"openapi":"3.1.0","info":{"title":"RestCountries.NET API","description":"Web API version 3.1 for managing country items, based on previous implementations from restcountries.eu and restcountries.com.","version":"v3.1"},"servers":[{"url":"https://restcountries.net"}],"auth":[],"paths":{"/v3.1/currency":{"get":{"description":"Search by currency.","operationId":"LookupCountryByCurrency","parameters":[{"name":"currency","in":"query","description":"The currency to search for.","required":true,"schema":{"type":"string"}}],"responses":{"200":{"description":"Success","content":{"text/plain":{"schema":{"type":"string"}}}}}}}},"components":{"schemes":{}}}'
+            """;
+        AzureAIAgentFactory factory = new();
+
+        var agent = await factory.CreateAgentFromYamlAsync(this._kernel, text) as AzureAIAgent;
+
+        await InvokeAgentAsync(agent, "Use code to determine the values in the Fibonacci sequence that that are less then the value of 101?");
+    }
+
+    [Fact]
+    public async Task AzureAIAgentWithLocalFunctionsAsync()
+    {
+        var text =
+            """
+            type: azureai_agent
+            name: RestaurantHost
+            instructions: Answer questions about the menu.
+            description: This agent answers questions about the menu.
+            model:
+              id: gpt-4o-mini
+              options:
+                temperature: 0.4
+                function_choice_behavior:
+                  type: auto
+                  functions:
+                    - MenuPlugin.GetSpecials
+                    - MenuPlugin.GetItemPrice
+            """;
+        AzureAIAgentFactory factory = new();
+
+        KernelPlugin plugin = KernelPluginFactory.CreateFromType<MenuPlugin>();
+        this._kernel.Plugins.Add(plugin);
+
+        var agent = await factory.CreateAgentFromYamlAsync(this._kernel, text) as AzureAIAgent;
+
+        await InvokeAgentAsync(agent, "What is the special soup and how much does it cost?");
+    }
+
+    #region private
+    private readonly Kernel _kernel;
+
+    /// <summary>
+    /// Invoke the agent with the user input.
+    /// </summary>
+    private async Task InvokeAgentAsync(AzureAIAgent? agent, string input)
+    {
+        // Create a thread for the agent conversation.
+        AgentThread thread = await this.AgentsClient.CreateThreadAsync(metadata: SampleMetadata);
+
+        // Respond to user input
+        try
+        {
+            await InvokeAsync(input);
+        }
+        finally
+        {
+            await this.AgentsClient.DeleteThreadAsync(thread.Id);
+            await this.AgentsClient.DeleteAgentAsync(agent!.Id);
```

Comment:
If agent can be null, I would check for it first or change the `AzureAIAgent?` to `AzureAIAgent`.

```suggestion
            await this.AgentsClient.DeleteAgentAsync(agent!.Id);
```